android/src/main/java/sqip/react/CardEntryModule.java

```diff
@@ -63,31 +80,62 @@ public CardEntryModule(final ReactApplicationContext reactContext) {
     reactContext.addActivityEventListener(new BaseActivityEventListener() {
       @Override
       public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
-        CardEntry.handleActivityResult(data, new Callback<CardEntryActivityResult>() {
-          @Override public void onResult(final CardEntryActivityResult cardEntryActivityResult) {
-            // React Native UI doesn't know the context of fade_out animation
-            // so that the next action from react native can be triggered too soon before
-            // card entry activity is closed completely.
-            // So this is a workaround to delay the callback until animation finished.
-            long delayDurationMs = readCardEntryCloseExitAnimationDurationMs();
-            mainLooperHandler.postDelayed(new Runnable() {
-              @Override
-              public void run() {
-                if (cardEntryActivityResult.isCanceled()) {
-                  getDeviceEventEmitter().emit("cardEntryCancel", null);
-                } else if (cardEntryActivityResult.isSuccess()) {
-                  getDeviceEventEmitter().emit("cardEntryComplete", null);
-                }
+        if (requestCode == CardEntry.DEFAULT_CARD_ENTRY_REQUEST_CODE) {
+          CardEntry.handleActivityResult(data, new Callback<CardEntryActivityResult>() {
+            @Override public void onResult(final CardEntryActivityResult cardEntryActivityResult) {
+              if (cardEntryActivityResult.isSuccess() && CardEntryModule.this.contact != null) {
+                cardResult = cardEntryActivityResult.getSuccessValue();
+                String paymentSourceId = cardResult.getNonce();
+                VerificationParameters verificationParameters = new VerificationParameters(paymentSourceId, CardEntryModule.this.buyerAction, CardEntryModule.this.squareIdentifier, CardEntryModule.this.contact);
```

Comment:
I only saw lint errors for javascript code. But updated anyway.

Reply:
open the project in Android Studio, then run code analysis, it will show you the lint errors, no need to fix all of them, just find the obvious java style errors.